clar2wasm/src/words/equal.rs

```diff
@@ -88,6 +89,179 @@ impl Word for IsEq {
+        };
+
+        // TODO return when size = 0
+
+        // Store the end of the sequence into a local.
+        let elem_size = get_type_size(elem_ty);
+        let end_offset = generator.module.locals.add(ValType::I32);
+        builder
+            .local_get(offset)
+            .i32_const(seq_size as i32 * elem_size)
+            .binop(BinaryOp::I32Add)
+            .local_set(end_offset);
+
+        // Traverse the item, leaving it on top of the stack.
+        let item = args.get_expr(1)?;
+        generator.traverse_expr(builder, item)?;
+        // STACK: [item]
+
+        // Get the type of the item expression
+        let item_ty = generator
+            .get_expr_type(item)
+            .expect("index_of item expression must be typed")
+            .clone();
+
+        // Store the item into a local.
+        let item_locals = generator.save_to_locals(builder, &item_ty, true);
+        // STACK: []
+
+        // Create and store an index into a local.
+        let index = generator.module.locals.add(ValType::I32);
+        builder.i32_const(0);
+        // STACK: [0]
+        builder.local_set(index);
+        // STACK: []
+
+        // Code block label to be used in the loop.
+        let block_id = builder.id();
+
+        // Loop through the sequence.
+        let loop_body_ty = InstrSeqType::new(
+            &mut generator.module.types,
+            &[],
+            &[ValType::I32, ValType::I64, ValType::I64],
+        );
+        builder.loop_(loop_body_ty, |loop_| {
+            // Loop label.
+            let loop_id = loop_.id();
+
+            // Load an element from the sequence, at offset position,
+            // and push it onto the top of the stack.
+            let elem_size = generator.read_from_memory(loop_, offset, 0, elem_ty);
+            // STACK: [element]
+
+            // Store the current sequence element into a local.
+            let elem_locals = generator.save_to_locals(loop_, elem_ty, true);
+            // STACK: []
+
+            // Check item and element equality.
+            // And push the result of the comparison onto the top of the stack.
+            let _res = wasm_equal(
+                &item_ty,
+                elem_ty,
+                generator,
+                loop_,
+                &item_locals,
+                &elem_locals,
+            );
+            // STACK: [wasm_equal_result]
+
+            // Store the result into a local.
+            let result = generator.module.locals.add(ValType::I32);
+            loop_.local_tee(result);
+            // STACK: [wasm_equal_result]
+
+            // Build the stack to be returned.
+            loop_
+                .i32_const(1)
+                .binop(BinaryOp::I32Eq)
+                // STACK: [(0 | 1)]
+                .local_get(index)
+                .unop(UnaryOp::I64ExtendUI32)
+                .i64_const(0);
+            // STACK: [(0 | 1), index_lo, index_hi]
```

Comment:
You can save some instructions by only building this result if you are going to be returning. Maybe something like this pseudo-code instead:
```
loop:
  read element
  save to locals
  call wasm_equal
  if true:
    loop_.i32_const(1).local_get(index).i64_const(0)
  else:
    increment offset
    if offset >= end_offset:
      loop_.i32_const(0).i64_const(0).i64_const(0)
    else:
      increment index
      br loop_
  ```